### PR TITLE
Fixup internal structure

### PR DESCRIPTION
**Pull request by sgnn7**
#### What does this PR do (include background context, if relevant)?

Improved internal structure of our broker:
- `internal/pkg/*` -> `internal/`
- `internal/app/secretless/*` -> `internal/`
- `internal/app/summon/*` -> `internal/summon/`

#### What ticket does this PR close?
N/A - out-of-band

#### Where should the reviewer start?
[Jenkins Build](https://jenkins.conjur.net/job/cyberark--secretless-broker/job/fixup-internal-structure/)

#### What is the status of the manual tests?
Have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/manual/keychain_provider)
- [ ] Manually run the [K8s demo](https://github.com/cyberark/secretless-broker/tree/master/demos/k8s-demo) (which requires changing the image the demo uses to the local build of Secretless)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

If this feature does not have any/sufficent automated tests, have you created/updated a folder in `test/manual` that includes:
- [ ] An updated README with instructions on how to manually test this feature
- [ ] Utility `start` and `stop` scripts to spin up and tear down the test environments
- [ ] A `test` script to run some basic manual tests (optional; if does not exist, the README should have detailed instructions)

### CODE REVIEW

**Comment by sgnn7** [comment]
PS: All failures from CC are old afaict

File: internal/handlers/mysql/authentication_handshake.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package mysql
 
 import (
-	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/mysql/protocol"
-	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/ssl"
+	"github.com/cyberark/secretless-broker/internal/handlers/mysql/protocol"
+	"github.com/cyberark/secretless-broker/internal/handlers/ssl"
 	"net"
```
Comment (doodlesbykumbi) [inline internal/handlers/mysql/authentication_handshake.go]: sort imports

File: internal/handlers/mysql/connection.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package mysql
 
 import (
-	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/mysql/protocol"
+	"github.com/cyberark/secretless-broker/internal/handlers/mysql/protocol"
```
Comment (doodlesbykumbi) [inline internal/handlers/mysql/connection.go]: sort imports

File: internal/providers/awssecrets/provider.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/secretsmanager"
```
Comment (doodlesbykumbi) [inline internal/providers/awssecrets/provider.go]: sort imports

File: internal/summon/command/command.go [hunk 4]
```diff
@@ -9,10 +9,10 @@ import (
 	"github.com/codegangsta/cli"
 	"github.com/cyberark/summon/secretsyml"
 
-	"github.com/cyberark/secretless-broker/internal/app/secretless"
-	"github.com/cyberark/secretless-broker/internal/pkg/plugin"
+	"github.com/cyberark/secretless-broker/internal"
+	"github.com/cyberark/secretless-broker/internal/plugin"
```
Comment (doodlesbykumbi) [inline internal/summon/command/command.go]: no need for this newline

File: test/aws_secrets_provider/aws_secrets_provider_test.go [hunk 5]
```diff
@@ -3,8 +3,8 @@ package main
 import (
 	"testing"
 
-	plugin_v1 "github.com/cyberark/secretless-broker/internal/app/secretless/plugin/v1"
-	"github.com/cyberark/secretless-broker/internal/app/secretless/providers"
+	plugin_v1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
```
Comment (doodlesbykumbi) [inline test/aws_secrets_provider/aws_secrets_provider_test.go]: sort imports

File: test/conjur/conjur_provider_test.go [hunk 6]
```diff
@@ -6,8 +6,8 @@ import (
 
 	. "github.com/smartystreets/goconvey/convey"
 
-	pluginV1 "github.com/cyberark/secretless-broker/internal/app/secretless/plugin/v1"
-	"github.com/cyberark/secretless-broker/internal/app/secretless/providers"
+	pluginV1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
+	"github.com/cyberark/secretless-broker/internal/providers"
 
 	_ "github.com/joho/godotenv/autoload"
```
Comment (doodlesbykumbi) [inline test/conjur/conjur_provider_test.go]: sort imports

File: test/kubernetes_provider/kubernetes_provider_test.go [hunk 7]
```diff
@@ -3,14 +3,15 @@ package main
...
+
 	. "github.com/smartystreets/goconvey/convey"
 
-	plugin_v1 "github.com/cyberark/secretless-broker/internal/app/secretless/plugin/v1"
-	"github.com/cyberark/secretless-broker/internal/app/secretless/providers"
-	"github.com/cyberark/secretless-broker/internal/app/secretless/providers/kubernetessecrets"
+	plugin_v1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
+	"github.com/cyberark/secretless-broker/internal/providers"
+	"github.com/cyberark/secretless-broker/internal/providers/kubernetessecrets"
 	"k8s.io/api/core/v1"
```
Comment (doodlesbykumbi) [inline test/kubernetes_provider/kubernetes_provider_test.go]: sort imports and remove line 8

File: test/summon2/summon2_cmd_test.go [hunk 8]
```diff
@@ -6,7 +6,7 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/cyberark/secretless-broker/internal/app/summon/command"
+	"github.com/cyberark/secretless-broker/internal/summon/command"
 
 	"bufio"
```
Comment (doodlesbykumbi) [inline test/summon2/summon2_cmd_test.go]: sort imports

File: test/summon2/summon2_run_test.go [hunk 9]
```diff
@@ -9,8 +9,8 @@ import (
 
 	"github.com/cyberark/summon/secretsyml"
 
-	plugin_v1 "github.com/cyberark/secretless-broker/internal/app/secretless/plugin/v1"
-	"github.com/cyberark/secretless-broker/internal/app/summon/command"
+	plugin_v1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
+	"github.com/cyberark/secretless-broker/internal/summon/command"
 
 	. "github.com/smartystreets/goconvey/convey"
```
Comment (doodlesbykumbi) [inline test/summon2/summon2_run_test.go]: sort imports

File: test/vault_provider/vault_summon_test.go [hunk 10]
```diff
@@ -6,7 +6,7 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/cyberark/secretless-broker/internal/app/summon/command"
+	"github.com/cyberark/secretless-broker/internal/summon/command"
 	. "github.com/smartystreets/goconvey/convey"
 
 	_ "github.com/joho/godotenv/autoload"
```
Comment (doodlesbykumbi) [inline test/vault_provider/vault_summon_test.go]: sort imports

File: test/vault_provider/vault_provider_test.go [hunk 11]
```diff
@@ -5,8 +5,8 @@ import (
 
 	. "github.com/smartystreets/goconvey/convey"
 
-	plugin_v1 "github.com/cyberark/secretless-broker/internal/app/secretless/plugin/v1"
-	"github.com/cyberark/secretless-broker/internal/app/secretless/providers"
+	plugin_v1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
+	"github.com/cyberark/secretless-broker/internal/providers"
 
 	_ "github.com/joho/godotenv/autoload"
```
Comment (doodlesbykumbi) [inline test/vault_provider/vault_provider_test.go]: sort imports

**Pull request merged**